src/master-playlist-controller.js

```diff
@@ -431,6 +433,22 @@ export class MasterPlaylistController extends videojs.EventTarget {
         return;
       }
 
+      for(var i = 0; i < updatedPlaylist.segments; i++) {
```

Comment:
We should move these checks out of the `loadedplaylist` listener and into `loadedmetadata` listener. `laodedplaylist` will be fired for every playlist refresh in a live stream. 

Also, since `loadedmetadata` is only fired once for the first media playlist we load, moving this there would remove the need to have extra state for `isAes_ and `isFmp4`.

I also think it would be a good idea to move these loops into helper functions either on the `PlaylistLoader` or in `playlist.js`